parser/parser.go

```diff
@@ -328,3 +550,40 @@ func isValidCommand(cmd string) bool {
 		return false
 	}
 }
+
+func expandPathImpl(path string, currentUserFunc func() (*user.User, error), lookupUserFunc func(string) (*user.User, error)) (string, error) {
```

Comment:
Could we put this in `expandPath` directly?

Reply:
This is split out so we can pass in the mock functions, otherwise any call to expandPath() would have to include passing in the functions from `os` which is less than ideal.